app/src/main/java/com/example/sam/bucketlist/api/APIManager.java

```diff
@@ -0,0 +1,146 @@
+package com.example.sam.bucketlist.api;
+
+import android.app.Activity;
+import android.app.usage.NetworkStats;
+import android.content.Context;
+import android.media.session.MediaSession;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+
+import com.example.sam.bucketlist.R;
+import com.example.sam.bucketlist.models.BucketListFields;
+import com.example.sam.bucketlist.models.BucketListPost;
+import com.example.sam.bucketlist.models.LoginFields;
+import com.example.sam.bucketlist.models.UserFields;
+import com.example.sam.bucketlist.service.UserClient;
+import com.example.sam.bucketlist.views.bucketlists.BucketListAdapter;
+
+import org.json.JSONException;
+
+import java.util.ArrayList;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Class APIManager Application Operations
+ */
+
+public class APIManager {
+
+    ArrayList<BucketListFields> bucketListData = new ArrayList<>();
+
+
+    Retrofit.Builder builder = new Retrofit.Builder()
+            .baseUrl("http://10.0.2.2:5000/api/v1/")
+            .addConverterFactory(GsonConverterFactory.create());
+
```

Comment:
I reccomend that you 

- Define the interceptor and add authentication headers
- Add the interceptor to OkHttpClient
- Set the custom client when building adapter

The interceptor is used to add token to the request as a authorization in the header. 
After adding the token to the request in the interceptor class then use it as shown below instead of passing token the as param in all requests. Eg:

- `OkHttpClient.Builder okHttpClient = new OkHttpClient().newBuilder();`
- `okHttpClient.addInterceptor(YourInterceptor()).build();`
- `builder.client(okHttpClient)` this is retrofit object in Line 40

[Check it here futurestud](https://futurestud.io/tutorials/retrofit-getting-started-and-android-client) or [CodePath](https://guides.codepath.com/android/consuming-apis-with-retrofit#retrofit-and-authentication)